Fix Gnulib test failures showing up on ARMv7:

  https://issues.guix.gnu.org/49459

This is a backport of this Gnulib commit:

  commit 175e0bc72808d564074c4adcc72aeadb74adfcc6
  Author: Paul Eggert <eggert@cs.ucla.edu>
  Date:   Thu Aug 27 17:52:58 2020 -0700

      perror, strerror_r: remove unportable tests

      Problem reported by Florian Weimer in:
      https://lists.gnu.org/r/bug-gnulib/2020-08/msg00220.html
      * tests/test-perror2.c (main):
      * tests/test-strerror_r.c (main): Omit unportable tests.

diff --git a/tests/test-perror2.c b/tests/test-perror2.c
index 1d14eda7be..c6214dd25c 100644
--- a/gnulib-tests/test-perror2.c
+++ b/gnulib-tests/test-perror2.c
@@ -79,9 +79,6 @@ main (void)
     errno = -5;
     perror ("");
     ASSERT (!ferror (stderr));
-    ASSERT (msg1 == msg2 || msg1 == msg4 || STREQ (msg1, str1));
-    ASSERT (msg2 == msg4 || STREQ (msg2, str2));
-    ASSERT (msg3 == msg4 || STREQ (msg3, str3));
     ASSERT (STREQ (msg4, str4));
 
     free (str1);
diff --git a/tests/test-strerror_r.c b/tests/test-strerror_r.c
index b11d6fd9f6..c1dbcf837b 100644
--- a/gnulib-tests/test-strerror_r.c
+++ b/gnulib-tests/test-strerror_r.c
@@ -165,9 +165,6 @@ main (void)
 
     strerror_r (EACCES, buf, sizeof buf);
     strerror_r (-5, buf, sizeof buf);
-    ASSERT (msg1 == msg2 || msg1 == msg4 || STREQ (msg1, str1));
-    ASSERT (msg2 == msg4 || STREQ (msg2, str2));
-    ASSERT (msg3 == msg4 || STREQ (msg3, str3));
     ASSERT (STREQ (msg4, str4));
 
     free (str1);
